pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -524,7 +524,17 @@ def _should_check_val_fx(self, batch_idx: int, is_last_batch: bool) -> bool:
         if isinstance(self.trainer.limit_train_batches, int) and is_infinite_dataset:
             is_val_check_batch = (batch_idx + 1) % self.trainer.limit_train_batches == 0
         elif self.trainer.val_check_batch != float("inf"):
-            is_val_check_batch = (batch_idx + 1) % self.trainer.val_check_batch == 0
+            # first we check if `check_val_every_n_epoch is `None`, which means
+            # that we run a validation loop after n global steps (n is taken from the
+            # Trainer argument `val_check_interval`)
+            if self.trainer.check_val_every_n_epoch is None:
+                is_val_check_batch = self.trainer.global_step % self.trainer.val_check_batch == 0
```

Comment:
`batches_that_stepped`. Good catch.

Reply:
asking because if I am not wrong global_step is always a multiple of number of optimizers in an ideal case so if someone wants to validate after every `N` steps they have to set the value as `N * number of optimizers`